docker/daemon/daemon_dest.go

```diff
@@ -91,9 +91,11 @@ func imageLoadGoroutine(ctx context.Context, c *client.Client, reader *io.PipeRe
 }
 
 // Close removes resources associated with an initialized ImageDestination, if any.
-func (d *daemonImageDestination) Close() {
+func (d *daemonImageDestination) Close() error {
+	var err error
 	if !d.committed {
 		logrus.Debugf("docker-daemon: Closing tar stream to abort loading")
+		err = errors.New("Aborting upload, daemonImageDestination closed without a previous .Commit()")
```

Comment:
This is not an error in `.Close()`; if the caller decides to abort without a `.Commit()`, and the upload is successfully aborted, `.Close()` should indicate success.